pkg/reconciler/managed/reconciler.go

```diff
@@ -1073,16 +1087,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (resu
 		}
 	}
 
+	reconcileAfter := r.pollInterval + time.Duration((rand.Float64()-0.5)*2*float64(r.pollJitter)) //#nosec G404 -- no need for secure randomness
```

Comment:
To provide more flexibility to provider developers it might make sense to not hardcode the jitter algorithm and instead provide a hook via a delegate that can be passed via an option:

```suggestion
	reconcileAfter := r.modifyRequeueAfter(managed, r.pollInterval)
```

In the future this might also be a way for providers to individually decide on the requeue time for individual resources.

Reply:
How do you feel about doing both, and having the `modifyRequeueAfter` hook accept both the configured poll interval and computed jitter (with the default implementation being to add them together)?

I think the library should strongly encourage provider developers to include support for jitter as a relatively well-establish best practice (whether they hardcode the amount or expose a flag), rather than it being something they each have to remember to do when they discover the hook exists?

Reply:
Well, you *do* can provide a default Jitter with the `NewReconciler`. I just would prefer to have the Jitter algorithm (or any kind of "delay" mechanic) not directly implemented within the reconciler but provide it via some form of plugin mechanism.

Reply:
What do you think of this approach, with the WithJitter option implemented as a WithPollIntervalHook?

Reply:
Looks good to me. However, is support for multiple hooks really necessary? I think it makes the hook mechanic more complex then it needs to be. And I spontaneously cannot think of a scenario where two hooks are dependent on each other (unlike overwriting).

Reply:
The main situation I'm thinking of is letting somebody do resource-based customisation, and have jitter applied on top, without them having to write(/copy-and-paste) their own jitter logic. Happy to change it to a single hook if you think that's insufficient justification.

Reply:
If you would have multiple hooks they would be called for every resource anyway and since it is all inside a single controller I think one hook should be fine. If people want to apply resource-based jitter then I assume it would suffice if they only implement one hook that can decide on it.

Reply:
Okay, switched it to a single hook